internal/sql/parser/select.go

```diff
@@ -154,3 +160,24 @@ func (p *Parser) parseGroupBy() (expr.Expr, error) {
 	e, err := p.ParseExpr()
 	return e, err
 }
+
+func (p *Parser) parseUnion() (*statement.StreamStmt, bool, error) {
+	// Only UNION ALL is supported for the moment
+	if ok, err := p.parseOptional(scanner.UNION, scanner.ALL); !ok || err != nil {
```

Comment:
This will allow `UNION` to go through:

```
// parseOptional parses a list of consecutive tokens. If the first token is not
// present, it unscans and return false. If the first is present, all the others
// must be parsed otherwise an error is returned.
```

Reply:
I don't follow why it would go through:

- If `UNION` is not present, the doc says that it will unscan, so doing nothing.
- If `UNION` is present, but `ALL` isn't, it says that an error will be returned. 

I wrote a unit test especially for this one: https://github.com/genjidb/genji/pull/408/files#diff-3d23dd0084a7b406c7e9e514fc99db24c108540d786a75ab0b0da7fea1ff0450R63-R68

```
genji> select * from foo union select * from bar;
found SELECT, expected ALL at line 1, char 25
genji>
``` 

@asdine Am I missing your point here?

Reply:
My bad, this is what happens when you review outdoors on your phone 🤦‍♂️

Reply:
Hahah np! I The mobile app is really getting better these days. I have definitively done that mistake more than once, it's so tempting to do it on the go 😺 !